internal/backend/vars/vars.go

```diff
@@ -2,34 +2,90 @@ package vars
 
 import (
 	"context"
+	"fmt"
 
 	"go.uber.org/zap"
 
 	"go.autokitteh.dev/autokitteh/internal/backend/db"
+	"go.autokitteh.dev/autokitteh/internal/backend/secrets"
 	"go.autokitteh.dev/autokitteh/sdk/sdkservices"
 	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
 )
 
-type vars struct{ db db.DB }
+type vars struct {
+	db      db.DB
+	secrets secrets.Secrets
+	z       *zap.Logger
+}
+
+func New(z *zap.Logger, db db.DB, secrets secrets.Secrets) sdkservices.Vars {
+	return &vars{db: db, z: z, secrets: secrets}
+}
 
-func New(z *zap.Logger, db db.DB) sdkservices.Vars {
-	return &vars{db: db}
+func varSecretKey(secret sdktypes.Var) string {
+	return fmt.Sprintf("%s/%s", secret.ScopeID().AsID().UUIDValue().String(), secret.Name())
 }
 
 func (v *vars) Set(ctx context.Context, vs ...sdktypes.Var) error {
+	for i, va := range vs {
+		if va.IsSecret() {
+			key := varSecretKey(va)
+			if err := v.secrets.Set(ctx, key, va.Value()); err != nil {
+				//TODO: handle delete secrets that were already created
```

Comment:
ticket

Reply:
https://linear.app/autokitteh/issue/ENG-817/handle-dangling-secrets

this is a new kind of task, we should have back office maintenance jobs to clean up dangling secrets